hr_holidays_hour/models/hr_holidays.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import datetime
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError, Warning
+
+
+class HrHolidays(models.Model):
+    _inherit = "hr.holidays"
+
+    @api.onchange('employee_id')
+    def onchange_holiday_employee(self):
+        self.department_id = None
+        self.number_of_hours_temp = 0.0
+        if self.employee_id:
+            self._set_number_of_hours_temp()
+            self.department_id = self.employee_id.department_id
+
+    @api.onchange('date_from', 'date_to')
+    def onchange_date(self):
+        # Check in context what form is open: add or remove
+        if self.env.context.get('default_type', '') == 'add':
+            return
+
+        self._check_dates()
+        self._check_employee()
+        self._set_number_of_hours_temp()
+
+    @api.multi
+    def _set_number_of_hours_temp(self):
+        self.ensure_one()
+        from_dt = self._compute_datetime(self.date_from)
+        to_dt = self._compute_datetime(self.date_to)
+        work_hours = self._compute_work_hours(from_dt, to_dt)
+        self.number_of_hours_temp = work_hours
+
+    @api.model
+    def _compute_datetime(self, date):
+        dt = False
+        if date:
+            reference_date = fields.Datetime.context_timestamp(
+                self.env.user,
+                datetime(1990, 2, 8, 12)
+            )
+            dt = fields.Datetime.from_string(date)
+            tz_dt = fields.Datetime.context_timestamp(self.env.user, dt)
+            dt = dt + tz_dt.tzinfo._utcoffset
+            dt = dt - reference_date.tzinfo._utcoffset
```

Comment:
In Odoo the workschedule time is set as a float value, instead of a time value. This is the reason of such strange time-datas manipulations. The issue we experienced with our customers was in this case the requested leaves start in wintertime and end in summertime (and the other way round).